redisson/src/main/java/org/redisson/MapWriteBehindTask.java

```diff
@@ -100,7 +100,7 @@ private void flushTasks(Map<Object, Object> addedMap, List<Object> deletedKeys)
                 if (options.getWriter() != null) {
                     options.getWriter().write(addedMap);
                 } else {
-                    options.getWriterAsync().write(addedMap).toCompletableFuture().join();
+                    options.getWriterAsync().write(addedMap);
```

Comment:
join should be retained since write method is executed in a separate thread.

Reply:
Sorry, i don't quite understand this comment.

The line at 103 where you commented will call `RetryableMapWriterAsync::write`, which is executed in the same thread as `flushTasks`. Then, step in `RetryableMapWriterAsync:56` : `mapWriterAsync.write(addedMap).toCompletableFuture().join(); ` it has joined to wait for a separate thread.

So, do you mean I should let `RetryableMapWriterAsync:write` look like `return CompletableFuture.runAsync(() -> {//retry logic})`? Because only this way can help me understand this requested change.